### PR TITLE
[CFE-309]: Refactor(pages): refactor module details service files

### PR DESCRIPTION
**Pull request by Poafs1**
**NOTE**: New Module Info UI NOT Included in the PR. Will update it in next PR accordingly.

### CODE REVIEW

File: src/lib/services/move/module.ts [hunk 1]
```diff
@@ -191,3 +200,143 @@ export const getModules = async (
...
+  }));
+export type ModuleInfoResponse = z.infer<typeof zModuleInfoResponse>;
+
+export const getModuleInfo = async (
+  endpoint: string,
+  moduleName: string,
+  vmAddress: HexAddr
+): Promise<ModuleInfoResponse> =>
+  axios
+    .get(`${endpoint}/${encodeURIComponent(vmAddress)}/${moduleName}/info`)
```
Comment (evilpeach) [inline src/lib/services/move/module.ts:231]: apply `encodeURIComponent` to `moduleName`

**Review by songwongtp — CHANGES_REQUESTED** [review]

File: src/lib/pages/module-details/components/tables/history/PublishedEventsTableRow.tsx [hunk 2]
```diff
@@ -39,7 +39,7 @@ export const PolicyChanges = ({ history }: { history: ModuleHistory }) => {
           <Text variant="body2" color="text.dark">
             Changed from{" "}
             <Text as="span" fontWeight={700}>
-              {capitalize(history.previousPolicy)}
+              {capitalize(history.previousPolicy ?? undefined)}
```
Comment (songwongtp) [inline src/lib/pages/module-details/components/tables/history/PublishedEventsTableRow.tsx:42]: remove `?? undefined`

**Review by songwongtp — CHANGES_REQUESTED** [review]

File: src/lib/pages/module-details/components/tables/history/PublishedEventsTableRow.tsx (hunk 2 above)
Comment (Poafs1) [reply src/lib/pages/module-details/components/tables/history/PublishedEventsTableRow.tsx:42]: I also made a change from switch case to if/else. Please re-review. Thank!

File: src/lib/pages/module-details/index.tsx [hunk 3]
```diff
@@ -100,7 +83,7 @@ export const ModuleDetailsBody = ({ moduleData }: ModuleDetailsBodyProps) => {
         },
       });
     },
-    [moduleData.address, moduleData.moduleName, navigate, tab]
+    [moduleInfo, navigate, tab]
```
Comment (songwongtp) [inline src/lib/pages/module-details/index.tsx:86]: moduleInfo.address, moduleInfo.moduleName

**Review by songwongtp — CHANGES_REQUESTED** [review]

File: src/lib/services/move/module.ts [hunk 4]
```diff
@@ -191,3 +200,147 @@ export const getModules = async (
...
+
+const zModuleTxsResponse = z.object({
+  items: z.array(zTxsResponseItem),
+  total: z.number().nonnegative(),
+});
+export type ModuleTxsResponse = z.infer<typeof zModuleTxsResponse>;
+
+export const getModuleTxs = async (
+  endpoint: string,
+  address: HexAddr,
```
Comment (evilpeach) [inline src/lib/services/move/module.ts:266]: vmAddress

File: src/lib/services/move/module.ts [hunk 5]
```diff
@@ -191,3 +200,147 @@ export const getModules = async (
...
+  proposals: z.number().nonnegative().nullable(),
+});
+export type ModuleTableCountsResponse = z.infer<
+  typeof zModuleTableCountsResponse
+>;
+
+export const getModuleTableCounts = async (
+  endpoint: string,
+  moduleName: string,
+  vmAddress: HexAddr
```
Comment (evilpeach) [inline src/lib/services/move/module.ts:250]: Swap the line position of moduleName and vmAddress

File: src/lib/services/move/module.ts [hunk 6]
```diff
@@ -191,3 +200,147 @@ export const getModules = async (
...
+    recentPublishTransaction: parseTxHashOpt(
+      val.recent_publish_transaction ?? undefined
+    ),
+  }));
+export type ModuleInfoResponse = z.infer<typeof zModuleInfoResponse>;
+
+export const getModuleInfo = async (
+  endpoint: string,
+  moduleName: string,
+  vmAddress: HexAddr
```
Comment (evilpeach) [inline src/lib/services/move/module.ts:230]: ditto

File: src/lib/services/move/module.ts [hunk 7]
```diff
@@ -191,3 +200,147 @@ export const getModules = async (
...
+        },
+      }
+    )
+    .then(({ data }) => parseWithError(zModuleTxsResponse, data));
+
+const zModuleHistory = z
+  .object({
+    remark: zRemark,
+    upgrade_policy: z.nativeEnum(UpgradePolicy),
+    height: z.number(),
```
Comment (evilpeach) [inline src/lib/services/move/module.ts:289]: only positive number

File: src/lib/services/move/module.ts [hunk 8]
```diff
@@ -191,3 +200,147 @@ export const getModules = async (
...
+
+const zModuleHistoriesResponse = z.object({
+  items: z.array(zModuleHistory),
+  total: z.number().nonnegative(),
+});
+export type ModuleHistoriesResponse = z.infer<typeof zModuleHistoriesResponse>;
+
+export const getModuleHistories = async (
+  endpoint: string,
+  address: HexAddr,
```
Comment (evilpeach) [inline src/lib/services/move/module.ts:304]: vmAddress

File: src/lib/services/move/module.ts [hunk 9]
```diff
@@ -191,3 +200,147 @@ export const getModules = async (
...
+  items: z.array(zProposalsResponseItem),
+  total: z.number().nonnegative(),
+});
+export type ModuleRelatedProposalsResponse = z.infer<
+  typeof zModuleRelatedProposalsResponse
+>;
+
+export const getModuleRelatedProposals = async (
+  endpoint: string,
+  address: HexAddr,
```
Comment (evilpeach) [inline src/lib/services/move/module.ts:331]: ditto

File: src/lib/services/move/moduleService.ts [hunk 10]
```diff
@@ -430,3 +292,109 @@ export const useModules = (
     { ...options, retry: 1, refetchOnWindowFocus: false }
   );
 };
+
+export const useModuleInfo = (moduleName: string, vmAddress: HexAddr) => {
```
Comment (evilpeach) [inline src/lib/services/move/moduleService.ts:296]: swap order of arguments

File: src/lib/services/move/moduleService.ts [hunk 11]
```diff
@@ -430,3 +292,109 @@ export const useModules = (
...
+  const endpoint = useBaseApiRoute("modules");
+
+  return useQuery<ModuleInfoResponse>(
+    [CELATONE_QUERY_KEYS.MODULE_INFO, endpoint, moduleName, vmAddress],
```
Comment (evilpeach) [inline src/lib/services/move/moduleService.ts:300]: `[CELATONE_QUERY_KEYS.MODULE_INFO, endpoint, vmAddress, moduleName]`

File: src/lib/services/move/moduleService.ts [hunk 12]
```diff
@@ -430,3 +292,109 @@ export const useModules = (
...
+    {
+      retry: 1,
+      refetchOnWindowFocus: false,
+    }
+  );
+};
+
+export const useModuleTableCounts = (
+  moduleName: string,
+  vmAddress: HexAddr
```
Comment (evilpeach) [inline src/lib/services/move/moduleService.ts:311]: ditto

File: src/lib/services/move/moduleService.ts [hunk 13]
```diff
@@ -430,3 +292,109 @@ export const useModules = (
...
+) => {
+  const endpoint = useBaseApiRoute("modules");
+
+  return useQuery<ModuleTableCountsResponse>(
+    [CELATONE_QUERY_KEYS.MODULE_TABLE_COUNTS, endpoint, moduleName, vmAddress],
```
Comment (evilpeach) [inline src/lib/services/move/moduleService.ts:316]: ditto

File: src/lib/services/move/moduleService.ts [hunk 14]
```diff
@@ -430,3 +292,109 @@ export const useModules = (
...
+    async () => getModuleTableCounts(endpoint, moduleName, vmAddress),
+    {
+      retry: 1,
+      refetchOnWindowFocus: false,
+    }
+  );
+};
+
+export const useModuleTxs = (
+  address: HexAddr,
```
Comment (evilpeach) [inline src/lib/services/move/moduleService.ts:326]: vmAddress

File: src/lib/services/move/moduleService.ts [hunk 15]
```diff
@@ -430,3 +292,109 @@ export const useModules = (
...
+      isInitia,
+    ],
+    async () =>
+      getModuleTxs(endpoint, address, moduleName, limit, offset, isInitia),
+    { retry: 1, refetchOnWindowFocus: false, ...options }
+  );
+};
+
+export const useModuleHistories = (
+  address: HexAddr,
```
Comment (evilpeach) [inline src/lib/services/move/moduleService.ts:352]: vmAddress

File: src/lib/pages/module-details/components/ModuleInfo.tsx [hunk 16]
```diff
@@ -1,162 +1,38 @@
...
+  Nullable,
+  Option,
+  Proposal,
+  UpgradePolicy,
+} from "lib/types";
+
+import { ModuleInfoBody } from "./ModuleInfoBody";
+
+export interface ModuleInfoProps {
+  address: HexAddr;
```
Comment (evilpeach) [inline src/lib/pages/module-details/components/ModuleInfo.tsx:17]: vmAddress

File: src/lib/pages/module-details/components/tables/history/index.tsx [hunk 17]
```diff
@@ -1,29 +1,28 @@
...
-import { useModuleHistoriesByPagination } from "lib/services/move/moduleService";
-import type { Nullish, Option } from "lib/types";
+import { useModuleHistories } from "lib/services/move/moduleService";
+import type { HexAddr, Option } from "lib/types";
 
 import { PublishedEventsTable } from "./PublishedEventsTable";
 
 interface ModuleHistoryTableProps {
-  moduleId: Nullish<number>;
+  address: HexAddr;
```
Comment (evilpeach) [inline src/lib/pages/module-details/components/tables/history/index.tsx:14]: vmAddress

File: src/lib/pages/module-details/components/tables/ModuleRelatedProposalsTable.tsx [hunk 18]
```diff
@@ -42,28 +41,22 @@ export const ModuleRelatedProposalsTable = ({
...
-  } = useRelatedProposalsByModuleIdPagination(
-    moduleId,
+  } = useModuleRelatedProposals(
+    address,
+    moduleName,
     offset,
-    onViewMore ? 5 : pageSize
+    onViewMore ? 5 : pageSize,
+    {
+      onSuccess: (data) => {
```
Comment (evilpeach) [inline src/lib/pages/module-details/components/tables/ModuleRelatedProposalsTable.tsx:50]: remove explicit `{}`

File: src/lib/pages/module-details/components/tables/index.tsx [hunk 19]
```diff
@@ -26,11 +26,9 @@ export enum ModuleTablesTabIndex {
 interface ModuleTablesProps {
   address: HexAddr;
```
Comment (evilpeach) [inline src/lib/pages/module-details/components/tables/index.tsx:27]: vmAddress

File: src/lib/pages/module-details/index.tsx [hunk 20]
```diff
@@ -100,7 +84,7 @@ export const ModuleDetailsBody = ({ moduleData }: ModuleDetailsBodyProps) => {
...
```
Comment (evilpeach) [inline src/lib/pages/module-details/index.tsx:87]: Don't use object as dependencies. Let's use ` [moduleInfo.address, moduleInfo.moduleName, navigate, tab]`

File: src/lib/pages/module-details/index.tsx [hunk 21]
```diff
@@ -109,22 +93,16 @@ export const ModuleDetailsBody = ({ moduleData }: ModuleDetailsBodyProps) => {
...
       });
     }
-  }, [
-    router.isReady,
-    tab,
-    navigate,
-    moduleData.address,
-    moduleData.moduleName,
-  ]);
+  }, [router.isReady, tab, navigate, moduleInfo]);
```
Comment (evilpeach) [inline src/lib/pages/module-details/index.tsx:105]: ditto

File: src/lib/pages/module-details/index.tsx [hunk 22]
```diff
@@ -135,11 +113,11 @@ export const ModuleDetailsBody = ({ moduleData }: ModuleDetailsBodyProps) => {
   }, [router.isReady, tab, verificationLoading, verificationData]);
 
   useEffect(() => {
-    if (moduleTxsCount === 0) {
+    if (moduleTableCounts.txs === 0) {
       setOverviewTabIndex(ModuleTablesTabIndex.PublishedEvents);
       setTableTabIndex(ModuleTablesTabIndex.PublishedEvents);
     }
-  }, [moduleTxsCount]);
+  }, [moduleTableCounts]);
```
Comment (evilpeach) [inline src/lib/pages/module-details/index.tsx:120]: ditto

File: src/lib/pages/module-details/components/ModuleInfoBody.tsx [hunk 23]
```diff
@@ -0,0 +1,84 @@
...
+        />
+      </LabelText>
+    );
+  }
+
+  return <LabelText label="Created by">Genesis</LabelText>;
+};
+
+export const ModuleInfoBody = ({
+  address,
```
Comment (evilpeach) [inline src/lib/pages/module-details/components/ModuleInfoBody.tsx:43]: vmAddress

File: src/lib/pages/module-details/components/ModuleInfo.tsx [hunk 24]
```diff
@@ -1,162 +1,38 @@
...
+  upgradePolicy: UpgradePolicy;
+  transaction: Option<string>;
+  proposal: Nullable<Partial<Proposal>>;
+  isRepublished: boolean;
```
Comment (evilpeach) [inline src/lib/pages/module-details/components/ModuleInfo.tsx:21]: `proposal: Nullable<Pick<Proposal, "id" | "title">>;`

**Review by evilpeach — CHANGES_REQUESTED** [review]
1.

**Review by evilpeach — APPROVED** [review]

**Review by songwongtp — APPROVED** [review]

**Pull request merged**